File: client/ng-mes-auto/src/app/models/operator.ts
import {OperatorGroup} from './operator-group';
import {Permission} from './permission';

export class Operator {
  id: string;
  name: string;
  hrId: string;
  admin: boolean;
  roles: string[];
  groups: OperatorGroup[];
  permissions: Permission[];

  static assign(...sources: any[]): Operator {
    const result = Object.assign(new Operator(), ...sources);
    return result;
  }

  static toEntities(os: Operator[], entities?: { [id: string]: Operator }): { [id: string]: Operator } {
    return (os || []).reduce((acc, cur) => {
      acc[cur.id] = Operator.assign(cur);
      return acc;
    }, {...(entities || {})});
  }
}

File: mes-root/mes-auto-core/src/main/java/com/hengyi/japp/mes/auto/domain/Operator.java
package com.hengyi.japp.mes.auto.domain;

import com.fasterxml.jackson.annotation.JsonIgnore;
import com.github.ixtf.persistence.IOperator;
import com.hengyi.japp.mes.auto.domain.data.RoleType;
import lombok.*;

import javax.persistence.Cacheable;
import javax.persistence.Column;
import javax.persistence.Entity;
import javax.persistence.Id;
import javax.validation.constraints.NotBlank;
import java.util.Collection;

/**
 * @author jzb 2018-06-22
 */
@NoArgsConstructor
@ToString(onlyExplicitlyIncluded = true)
@EqualsAndHashCode(onlyExplicitlyIncluded = true)
@Cacheable
@Entity
public class Operator implements IOperator {
    @ToString.Include
    @EqualsAndHashCode.Include
    @Getter
    @Setter
    @Id
    private String id;
    @ToString.Include
    @Getter
    @Setter
    @Column
    @NotBlank
    private String name;
    @Getter
    @Setter
    @Column
    private String hrId;
    @Getter
    @Setter
    @Column
    private String oaId;
    @Getter
    @Setter
    @Column
    private String phone;
    @Getter
    @Setter
    @Column
    private boolean admin;
    @JsonIgnore
    @Getter
    @Setter
    @Column
    private Collection<OperatorGroup> groups;
    @JsonIgnore
    @Getter
    @Setter
    @Column
    private Collection<RoleType> roles;
    @JsonIgnore
    @Getter
    @Setter
    @Column
    private Collection<Permission> permissions;
    @JsonIgnore
    @Getter
    @Setter
    @Column
    private boolean deleted;

}
